schematools/validation.py

```diff
@@ -147,3 +150,60 @@ def validate(self) -> Iterator[ValidationError]:
                     f"Inferred PostgreSQL table name '{table_name}' is '{excess}' characters too "
                     f"long. Maximum table name length is '{MAX_TABLE_NAME_LENGTH}'.",
                 )
+
+
+class IdentPropRefsValidator(Validator):
+    """Validate that the identifier property refers to actual fields on the table definitions."""
+
+    def validate(self) -> Iterator[ValidationError]:
+        @dataclass
+        class DerivedField:
+            original: str
+            derived: str
+
+        for table in self.dataset.get_tables(include_nested=True, include_through=True):
+            identifiers = set(table.identifier)
+            table_fields = cast(Set[str], set(map(operator.attrgetter("id"), table.fields)))
+            if not identifiers.issubset(table_fields):
+                missing_fields = identifiers - table_fields
+                # The 'identifier' property is weird in that it is not exclusively defined in
+                # terms of literally defined fields on the table. For instance, given a relation:
+                #
+                #     "indicatorDefinitie": {
+                #       "type": "string",
+                #       "relation": "bbga:indicatorenDefinities",
+                #        "description": "De variabele in kwestie."
+                #     }
+                #
+                # 'identifier' can refer to this field as 'identifierDefinitionId' (mind the
+                # 'Id' postfix). Simply referring to this field (from 'identifier') as
+                # 'indicatorDefinitie', eg as:
+                #
+                #     "identifier": ["indicatorDefinitie", "jaar", "gebiedcode15"],
+                #
+                #  will NOT work. It has to be postfixed with 'Id', eg:
+                #
+                #     "identifier": ["indicatorDefinitieIdId", "jaar", "gebiedcode15"],
```

Comment:
Is it really `IdId`  (doubled)?

Reply:
Good catch, its just me copy/pasting/editing a little to aggressive. Will fix.